drivers/usbmisc/stusb4500.c

```diff
@@ -0,0 +1,614 @@
+
+struct stusb4500_dev_s
+{
+  FAR struct i2c_master_s *i2c;          /* I2C interface */
+  uint8_t addr;                          /* I2C address */
+  mutex_t devlock;                       /* Manages exclusive access */
+};
+
+/****************************************************************************
+ * Private Function prototypes
+ ****************************************************************************/
+
+#ifdef CONFIG_DEBUG_STUSB4500
+
+#endif
+static int stusb4500_open(FAR struct file *filep);
+static int stusb4500_close(FAR struct file *filep);
+static ssize_t stusb4500_read(FAR struct file *, FAR char *, size_t);
+static ssize_t stusb4500_write(FAR struct file *filep,
+                               FAR const char *buffer, size_t buflen);
+static int stusb4500_ioctl(FAR struct file *filep, int cmd,
+                           unsigned long arg);
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static const struct file_operations g_stusb4500ops =
+{
+  stusb4500_open,  /* open */
+  stusb4500_close, /* close */
+  stusb4500_read,  /* read */
+  stusb4500_write, /* write */
+  NULL,            /* seek */
+  stusb4500_ioctl, /* ioctl */
+  NULL,            /* mmap */
+  NULL,            /* truncate */
+  NULL             /* poll */
+};
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: stusb4500_getreg
+ *
+ * Description:
+ *   Read from an 8-bit STUSB4500 register
+ *
+ * Input Parameters:
+ *   priv   - pointer to STUSB4500 Private Structure
+ *   reg    - register to read
+ *   rxbuf  - pointer to receive data buffer
+ *   len    - receive data length
+ *
+ * Returned Value:
+ *   Returns positive register value in case of success, otherwise ERROR
+ *
+ ****************************************************************************/
+
+static int stusb4500_getreg(FAR struct stusb4500_dev_s *priv, uint8_t reg,
+                            uint8_t *rxbuf, uint8_t len)
+{
+  int ret = -EIO;
+  int retries;
+  struct i2c_msg_s msg[2];
+
+  DEBUGASSERT(priv);
+
+  msg[0].frequency = CONFIG_STUSB4500_I2C_FREQUENCY;
+  msg[0].addr      = priv->addr;
+  msg[0].flags     = I2C_M_NOSTOP;
+  msg[0].buffer    = &reg;
+  msg[0].length    = 1;
+
+  msg[1].frequency = CONFIG_STUSB4500_I2C_FREQUENCY;
+  msg[1].addr      = priv->addr;
+  msg[1].flags     = I2C_M_READ;
+  msg[1].buffer    = rxbuf;
+  msg[1].length    = len;
+
+  /* Perform the transfer */
+
+  for (retries = 0; retries < STUSB4500_I2C_RETRIES; retries++)
+    {
+      ret = I2C_TRANSFER(priv->i2c, msg, 2);
+      if (ret >= 0)
+        {
+#define BUFFER_SIZE 128
```

Comment:
Please move this macro to the right section of this file (top) or to stusb4500.h, also use STUSB4500_BUFFER_SIZE to avoid name collision